test/AspNetCoreModule.Test/Framework/InitializeTestMachine.cs

```diff
@@ -65,27 +83,135 @@ public class InitializeTestMachine : IDisposable
-                result = false;
+                return Path.Combine(Environment.ExpandEnvironmentVariables("%SystemDrive%") + @"\", "_ANCMTest");
             }
-            return result;
         }
+
+        private static string _globalTestFlags = null;
+        public static string GlobalTestFlags
+        {
+            get
+            {
+                if (_globalTestFlags == null)
+                {
+                    bool isElevated;
+                    WindowsIdentity identity = WindowsIdentity.GetCurrent();
+                    WindowsPrincipal principal = new WindowsPrincipal(identity);
+                    isElevated = principal.IsInRole(WindowsBuiltInRole.Administrator);
+
+                    // check if this test process is started with the Run As Administrator start option
+                    _globalTestFlags = Environment.ExpandEnvironmentVariables(ANCMTestFlagsEnvironmentVariable);
+                    _globalTestFlags = _globalTestFlags.ToLower();
+                    if (_globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()) && _globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                    {
+                        throw new ApplicationException(ANCMTestFlagsEnvironmentVariable + " can't be set with both "
+                            + ANCMTestFlags.UseIISExpressContext + " and " + ANCMTestFlags.UseFullIISContext);
+                    }
+
+                    // set default context with ANCMTestFlags_DefaultContext_IISExpressP0Only
+                    if (!isElevated
+                        || _globalTestFlags.Replace(ANCMTestFlagsEnvironmentVariable.ToLower(), "").Trim() == ""
+                        || _globalTestFlags.Contains(ANCMTestFlags.UseIISExpressContext.ToLower()))
+                    {
+                        _globalTestFlags = ANCMTestFlags_DefaultContext_IISExpressP0Only;
+                    }
+
+                    // if users set UseFullIIS, the default value should be overwritten with ANCMTestFlags_DefaultContext_FullIIS
+                    if (isElevated
+                        && _globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                    {
+                        _globalTestFlags = ANCMTestFlags_DefaultContext_FullIIS;
+                    }
+
+                    // adjust the default test context in run time to figure out wrong test context values
+                    if (!isElevated)
+                    {
+                        if (_globalTestFlags.Contains(ANCMTestFlags.UseFullIISContext.ToLower()))
+                        {
+                            _globalTestFlags = _globalTestFlags.Replace(ANCMTestFlags.UseFullIISContext.ToLower(), "");
```

Comment:
I'm not a fan of calling Replace like this. Maybe go back and build up the string based on flags?

Reply:
I agree but I will handle this in the future.